src/composer.rs

```diff
@@ -57,13 +58,15 @@ impl Composer for Barretenberg {
         pow2ceil(circuit_size + NUM_RESERVED_GATES)
     }
 
-    fn get_exact_circuit_size(&self, constraint_system: &ConstraintSystem) -> u32 {
+    fn get_exact_circuit_size(&self, constraint_system: &ConstraintSystem) -> Result<u32, Error> {
         let cs_buf = constraint_system.to_bytes();
 
-        unsafe { barretenberg_sys::composer::get_exact_circuit_size(cs_buf.as_slice().as_ptr()) }
+        Ok(unsafe {
+            barretenberg_sys::composer::get_exact_circuit_size(cs_buf.as_slice().as_ptr())
+        })
```

Comment:
Can we assign the return value from `get_exact_circuit_size`to a variable and then return that as `Ok(circuit_size)`? Having an unsafe block nested like this feels iffy.

Reply:
I'm pretty sure this is fine but I'll change it because the style doesn't matter to me